src/http-libwebsockets.c

```diff
@@ -573,11 +694,22 @@ static int callback_metrics(struct lws *wsi, enum lws_callback_reasons reason,
                 break;
             }
         }
-        int n = stats->current < metrics_length ? LWS_WRITE_HTTP : LWS_WRITE_HTTP_FINAL;
+
+        int alloc_cur = 0;
+        while (alloc_cur < allocator_metrics_length) {
+            if (write_allocator_metric(&position, end, &allocator_metrics[alloc_cur])) {
+                qd_log(hs->log, QD_LOG_DEBUG, "wrote allocator metric %lu of %lu", alloc_cur, allocator_metrics_length);
+                alloc_cur++;
+            } else {
+                qd_log(hs->log, QD_LOG_DEBUG, "insufficient space in buffer");
```

Comment:
I'm wondering about the log levels, `QD_LOG_DEBUG` for success seems high to me (I'd go with TRACE) and `QD_LOG_DEBUG` for the failure here could be too low (I'd give it INFO). But I did not examine qdrouterd logging comprehensively, so maybe I am just no properly attuned.

Reply:
Makes sense to me as well.